src/main/java/coms/w4156/moviewishlist/Services/ServiceForRepository.java

```diff
@@ -0,0 +1,55 @@
+package coms.w4156.moviewishlist.services;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.springframework.data.repository.CrudRepository;
+
+import coms.w4156.moviewishlist.models.ModelInterface;
+
+public abstract class ServiceForRepository<
+        ID extends Comparable<ID>,
+        Model extends ModelInterface<ID>,
+        Repository extends CrudRepository<Model, ID>
+> {
+    protected Repository repository;
+    
+    public List<Model> getAll() {
```

Comment:
The functions are the same as what was in WishlistService, but the types have been replaced with the type arguments above.

This makes the same methods work for all the different types of models.